src/transaction.js

```diff
@@ -39,43 +39,64 @@ export class Transaction {
     }
     this._networkPassphrase = networkPassphrase;
 
-    const txe = envelope.value();
+    const txEnvelope = envelope.value();
     let sourceAccount;
     this._envelopeType = envelope.switch();
     switch (this._envelopeType) {
       case xdr.EnvelopeType.envelopeTypeTxV0():
-        sourceAccount = txe.tx().sourceAccountEd25519();
+        sourceAccount = txEnvelope.tx().sourceAccountEd25519();
         break;
       case xdr.EnvelopeType.envelopeTypeTx():
-        sourceAccount = txe
+        sourceAccount = txEnvelope
           .tx()
           .sourceAccount()
           .ed25519();
         break;
+      case xdr.EnvelopeType.envelopeTypeTxFeeBump():
+        sourceAccount = txEnvelope
+          .tx()
+          .feeSource()
+          .ed25519();
+        break;
       default:
-        // TODO: Add support for FeeBumpTransaction
-        throw new Error('Invalid transaction.');
+        throw new Error('Unknown TransactionEnvelope');
```

Comment:
Obviously this should never happen. But if it did: can the error say something about the envelope that was received, e.g. the integer type

Reply:
Fixed